examples/coroutinesExample/src/main/java/io/realm/examples/coroutinesexample/util/NewsReaderFlowFactory.kt

```diff
@@ -0,0 +1,500 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.coroutinesexample.util
+
+import io.realm.*
+import io.realm.coroutines.FlowFactory
+import io.realm.rx.CollectionChange
+import io.realm.rx.ObjectChange
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.flowOf
+
+/**
+ * Similar to [io.realm.coroutines.RealmFlowFactory] but it will not emit the current value
+ * immediately. This is needed by Store to function properly or else it will receive updates with
+ * empty [RealmResults] that will make it think existing values for the current key are present.
+ */
+class NewsReaderFlowFactory : FlowFactory {
```

Comment:
Wouldn't a much cleaner solution be to just call the super methods and then drop or filter out the first element emitted?

Copying the entire internal implementation seems very error-prone in terms of maintaining it?

Reply:
Well spotted. I also thought about doing that but when I did this the updated factory (with the latest changes and such) wasn't in the source branch for this branch, so I copied it entirely from the flow factory branch I was simultaneously working on as temporary solution. It seems I forgot to clean it up 🙊